From c4dd9469ec6674872ba3c220325aef919b400523 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Thu, 14 Jun 2018 14:57:40 +0200
Subject: [PATCH 1/1] Fix permission check for auto translate
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is just display logic, so it is not a security issue
but just does not allow to use the feature to anybody.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 weblate/trans/views/basic.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/weblate/trans/views/basic.py b/weblate/trans/views/basic.py
index 3b6be258a..8c4de7ae9 100644
--- a/weblate/trans/views/basic.py
+++ b/weblate/trans/views/basic.py
@@ -336,7 +336,7 @@ def show_project(request, project):
     )
 
     # Is user allowed to do automatic translation?
-    if not request.user.has_perm('translation.auto', obj):
+    if request.user.has_perm('translation.auto', obj):
         mass_state_form = MassStateForm(request.user, obj)
     else:
         mass_state_form = None
@@ -377,7 +377,7 @@ def show_component(request, project, component):
     last_changes = Change.objects.for_component(obj)[:10]
 
     # Is user allowed to do automatic translation?
-    if not request.user.has_perm('translation.auto', obj):
+    if request.user.has_perm('translation.auto', obj):
         mass_state_form = MassStateForm(request.user, obj)
     else:
         mass_state_form = None
@@ -423,13 +423,13 @@ def show_translation(request, project, component, lang):
     form = get_upload_form(request.user, obj)
 
     # Is user allowed to do automatic translation?
-    if not request.user.has_perm('translation.auto', obj):
+    if request.user.has_perm('translation.auto', obj):
         mass_state_form = MassStateForm(request.user, obj)
     else:
         mass_state_form = None
 
     # Is user allowed to do automatic translation?
-    if not request.user.has_perm('translation.auto', obj):
+    if request.user.has_perm('translation.auto', obj):
         autoform = AutoForm(obj, request.user)
     else:
         autoform = None
-- 
2.17.1

